packages/flutter/lib/src/foundation/convert_response.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+
+
+/// Efficiently converts the response body of an [HttpClientResponse] into a [Uint8List].
+/// 
+/// The future returned by [convert] will forward all errors emitted by [response].
+Future<Uint8List> convertResponse(HttpClientResponse response) {
```

Comment:
Since this is a global method, let's call it something more specific, like "consolidateHttpClientResponseBytes" or something.